fiftyone/server/files.py

```
@@ -0,0 +1,351 @@
+        finally:
+            await anyio.to_thread.run_sync(lambda: fh.close())
+    except FileOperationError:
+        raise
+    except OSError as e:
+        await anyio.to_thread.run_sync(lambda: _sync_try_remove(resolved_path))
+        if e.errno == 28:  # ENOSPC -- no space left on device
+            raise FileOperationError(
+                code=FileOperationErrorCode.STORAGE_FULL,
+                message="No space left on device.",
+                details={"path": resolved_path},
+            )
```

Ruff (0.15.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)